src/firmware/uapi/host/mod.rs

```diff
@@ -148,6 +148,46 @@ impl Firmware {
             },
         })
     }
+
+    /// Fetch the SNP Extended Configuration.
+    pub fn snp_get_ext_config(&mut self) -> Result<SnpExtConfig, Indeterminate<Error>> {
+        let mut config: SnpGetExtConfig = Default::default();
+        SNP_GET_EXT_CONFIG.ioctl(&mut self.0, &mut Command::from_mut(&mut config))?;
+
+        Ok(SnpExtConfig {
+            config_address: Some(config.config_address as *const SnpConfig),
+            certs_address: Some(config.certs_address as *const Chain),
+            certs_len: config.certs_len,
+        })
+    }
+
+    /// Set the SNP Extended Configuration.
+    pub fn snp_set_ext_config(
+        &mut self,
+        new_config: &SnpExtConfig,
+    ) -> Result<(), Indeterminate<Error>> {
+        let mut config: SnpSetExtConfig = Default::default();
+
+        match (new_config.certs_address, new_config.config_address) {
```

Comment:
Another option here (haven't tried it):

```rust
config.certs_address = new_config.certs_address.unwrap_or(0);
config.config_address = new_config.config_address.unwrap_or(0);
```

Another thing here which might be clearer: instead of initializing a `Default::default()` variable and then modifying it, maybe all this field-setting code should be in a `fn new(new_config: SnpExtConfig) -> Self` of `SnpSetExtConfig`. (but I'd follow what's done elsewhere in this crate if that's not the norm.)

Reply:
> Another thing here which might be clearer: instead of initializing a `Default::default()` variable and then modifying it, maybe all this field-setting code should be in a `fn new(new_config: SnpExtConfig) -> Self` of `SnpSetExtConfig`. (but I'd follow what's done elsewhere in this crate if that's not the norm.)

Yes, I think this would be fine, and would ultimately be the most clear.

Reply:
I adapted this a little bit. I couldn't use `unwrap_or(0)` as the data types didn't match. However, I used `unwrap_or(std::ptr::null)` and then used `as u64` after the unwrap.